### PR TITLE
feat!: do not auto pin rangeStrategy

### LINKED ISSUE #18304: Remove rangeStrategy=auto pinning
### What would you like Renovate to be able to do?

No longer auto pin if rangeStrategy=auto. Pinning should be fully opted into.

### If you have any ideas on how this should be implemented, please tell us here.

In cases where rangeStrategy=auto, return 'replace' instead of 'pin'.

### Is this a feature you are interested in implementing yourself?

Yes

### PR DESCRIPTION
**Pull request by rarkins**
## Changes

Removes all logic where rangeStrategy=auto results in pinning.

## Context

Closes #18304 

BREAKING CHANGE: Dependencies are no longer automatically pinned, pinning must be opted into using rangeStrategy=pin

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/development/new-package-manager-template.md [hunk 1]
```diff
@@ -74,14 +74,6 @@
 - [ ] Supports range constraints (e.g `^1.0.0` or `1.x`)
 - [ ] No
 
-### Is this package manager used for applications, libraries, or both? If both, is there a way to tell which is which?
```
Comment (HonkingGoose) [inline docs/development/new-package-manager-template.md:77]: https://github.com/renovatebot/renovate/blob/a762677d6abafe60758e220ca26d73d0c10acfac/docs/usage/configuration-options.md?plain=1#L2298-L2329

Maybe you need to change things in these docs as well?

File: docs/usage/getting-started/installing-onboarding.md [hunk 2]
```diff
@@ -107,7 +107,7 @@ Sometimes Renovate detects that an override to these defaults is needed, and wil
 Please check the docs on this website for an exhaustive Configuration Reference.
 To help you get started, here are some of the most commonly changed (overridden) configuration settings:
 
-- **rangeStrategy**: By default (with zero config) it's `"replace"` but the `"config:base"` preset overrides it to `"auto"`. If you don't want to pin dependency versions and retain ranges, add the `":preserveSemverRanges"` preset to the `extends` array
+- **rangeStrategy**: By default (with zero config) it's `"replace"` but the `"config:base"` preset overrides it to `"auto"`. Some prefer `"bump"`.
```
[the remark this answers is not in the document]
Comment (rarkins) [reply docs/usage/getting-started/installing-onboarding.md:110]: I think better to leave that to the `rangeStrategy` documentation itself

**Review by HonkingGoose — APPROVED** [review]
Docs look good to me now. 😉

File: lib/modules/manager/bundler/range.ts [hunk 3]
```diff
@@ -1,22 +0,0 @@
...
- * For example in npm, when rangeStrategy is auto we:
- *  - Always pin "devDependencies"
- *  - Pin "dependencies" only if we detect that it's probably an app not a library
- *  - Always widen "peerDependencies"
- *
- * If this function is not present then the default 'replace' value will be used.
- *
- */
-export function getRangeStrategy(config: RangeConfig): RangeStrategy {
-  if (config.rangeStrategy === 'auto') {
```
Comment (viceice) [inline lib/modules/manager/bundler/range.ts:17]: is this case now handled somewhere else?

**Review by viceice — COMMENTED** [review]
otherwise LGTM

File: lib/modules/manager/bundler/range.ts (hunk 3 above)
Comment (rarkins) [reply lib/modules/manager/bundler/range.ts:17]: It was already redundant. Renovate falls back to replace if no getRangeStrategy is specified

**Review by viceice — APPROVED** [review]

**Pull request merged**